File: src/main/java/de/twometer/amongus/model/ClientSession.java
package de.twometer.amongus.model;

public class ClientSession extends Session<Player> {

    private int myPlayerId;

    public float taskProgress;

    public long lastEmergency = 0;

    public PlayerRole winners;

    public Sabotage currentSabotage = null;
    public String currentSabotageCode = "";
    public int currentSabotageDuration = 0;

    public ClientSession(String gameCode, int host) {
        super(gameCode, host);
    }

    public void setMyPlayerId(int myPlayerId) {
        this.myPlayerId = myPlayerId;
    }

    public Player getMyself() {
        for (Player player : getPlayers())
            if (player.id == myPlayerId)
                return player;
        throw new RuntimeException("Can't find myself.");
    }

    public int getMyPlayerId() {
        return myPlayerId;
    }

}

File: src/main/kotlin/de/twometer/amogus/client/ClientSession.kt
package de.twometer.amogus.client

import de.twometer.amogus.model.BaseSession
import de.twometer.amogus.model.Player
import de.twometer.amogus.model.PlayerRole

class ClientSession : BaseSession<Player>() {

    val myself get() = findPlayer(AmongUsClient.myPlayerId)!!
    val myselfOrNull get() = findPlayer(AmongUsClient.myPlayerId)
    val myselfIsHost get() = host == AmongUsClient.myPlayerId
    var taskProgress: Float = 0f
    var surveillanceActive: Boolean = false
    var winners: PlayerRole = PlayerRole.Crewmate

}